### PR TITLE
Make sure we write correctly to file log 

### PR DESCRIPTION
**Pull request by fvovan**
While testing puppet package I've noticed that:

Console logger prints:
````
root@debian-7-amd64-default:/home/vagrant# /usr/bin/node /usr/bin/cm-janus -c /etc/cm-janus/config.yaml
app debug WebSocket server on port 8210 started
app debug HTTP server on port 8800 started
app error [Error: Invalid file content "/tmp/fooooo'. Unexpected end of input]
app error [Error: Invalid file "/tmp/npm-12737-06351932'. EISDIR, read]
app error [Error: Invalid file "/tmp/npm-7796-2a936564'. EISDIR, read]
app error [Error: Invalid file "/tmp/npm-9111-5d1d6cb4'. EISDIR, read]
app error [Error: Invalid file "/tmp/vagrant-puppet'. EISDIR, read]
app error [Error: Invalid file "/tmp/proxy-cache'. EISDIR, read]
```

While file log looks like:
```
root@debian-7-amd64-default:/home/vagrant# tail -f /var/log/cm-janus/cm-janus.log 
app debug WebSocket server on port 8210 started
app debug HTTP server on port 8800 started
app error {}
app error {}
app error {}
app error {}
app error {}
app error {}
```

Seems javascript objects are not properly dumped. In other words add tests for logger.

### CODE REVIEW

**Comment by fvovan** [comment]
@tomaszdurka it was just another black magic JS feature related to non-enumerable properties.
In a glance `minilog.Stringifier()` uses `JSON.stringify()`  and  `JSON.stringify()` works only for  enumerable properties. But `Error` doesn't seem to have them. I tried to search for workaround and found some but it looks so bad..

File: test/unit/logger.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  before(function(done) {
+    fs.mkdir(globalTmpDir, function() {
+      logFilePath = tmpNameSync({dir: globalTmpDir, postfix: '.log'});
+      done();
+    });
+  });
+
+  after(function(done) {
+    rimraf(globalTmpDir, function(err) {
```
Comment (vogdb) [inline test/unit/logger.js]: you can cut it to
```js
rimraf(globalTmpDir, done);
```

File: test/unit/logger.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+var sinon = require('sinon');
+var Promise = require('bluebird');
+var path = require('path');
+var fs = require("fs");
+var tmpNameSync = require('tmp').tmpNameSync;
+var rimraf = require('rimraf');
+
+var Logger = require('../../lib/logger');
+
+var globalTmpDir = path.join(__dirname, '/tmp');
```
Comment (vogdb) [inline test/unit/logger.js]: lets move it inside `describe` please

**Comment by vogdb** [comment]
the rest looks good

File: test/unit/logger.js (hunk 2 above)
Comment (vogdb) [inline test/unit/logger.js]: I mean `globalTmpDir` definition

**Pull request merged**